docker/get-component-version.sh

```diff
@@ -1,21 +1,35 @@
-#!/bin/bash
-
-#  Copyright (C) 2018-2021 LEIDOS.
-# 
-#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
-#  use this file except in compliance with the License. You may obtain a copy of
-#  the License at
-# 
-#  http://www.apache.org/licenses/LICENSE-2.0
-# 
-#  Unless required by applicable law or agreed to in writing, software
-#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
-#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
-#  License for the specific language governing permissions and limitations under
-#  the License.
-
-cd "$(dirname "$0")"
-cd ..
-COMPONENT_TAG_PREFIX="${PWD##*/}"
-git describe --all --match="$COMPONENT_TAG_PREFIX*" --always --dirty="-SNAPSHOT" | awk -F "/" '{print $NF}' | sed "s/$COMPONENT_TAG_PREFIX\_//"
-
+#  Copyright (C) 2018-2024 LEIDOS.
+# 
+#  Licensed under the Apache License, Version 2.0 (the "License"); you may not
+#  use this file except in compliance with the License. You may obtain a copy of
+#  the License at
+# 
+#  http://www.apache.org/licenses/LICENSE-2.0
+# 
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#  WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#  License for the specific language governing permissions and limitations under
+#  the License.
+
+# Change to the directory containing the script
+cd "$(dirname "$0")"
+
+# Move up one directory (likely to the root of the project)
+cd ..
+
+# Get the last commit message
+COMMIT_MSG=$(git log -1 --pretty=%B)
+
+# Remove newlines and special characters, replace spaces with underscores
+VERSION=$(echo "$COMMIT_MSG" | tr -d '\n' | tr -dc '[:alnum:][:space:]' | tr '[:space:]' '_')

```

Comment:
so the version is generated from the commit message, which works well for general versioning should we consider appending a suffix like -focal or -jammy to the VERSION variable depending on which image is being built (focal or jammy)?
some thing like VERSION="${VERSION}-focal" ?

Reply:
The suffix is appended outside this function through variable called $tag_suffix. So it should be good.